_inc/lib/debugger/class-jetpack-debugger.php

```diff
@@ -225,94 +205,64 @@ public static function jetpack_debug_display_handler() {
-							</div>
-							<?php
-						} else {
-							$debug_info .= $test_name . ": PASS\r\n";
-						}
-						$debug_raw_info .= "\r\n\r\n" . $test_name . "\r\n" . esc_html( is_wp_error( $test_info['result'] ) ? $test_info['result']->get_error_message() : print_r( $test_info['result'], 1 ) ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
-						?>
-						<?php
-					endforeach;
-					$html = ob_get_clean();
-
-					if ( '' === trim( $html ) ) {
-						echo '<div class="jetpack-tests-succed">' . esc_html__( 'Your Jetpack setup looks a-okay!', 'jetpack' ) . '</div>';
+					if ( $cxntests->pass() ) {
+						echo '<div class="jetpack-tests-succeed">' . esc_html__( 'Your Jetpack setup looks a-okay!', 'jetpack' ) . '</div>';
+						$debug_info .= "All tests passed.\r\n";
+						$debug_info .= print_r( $cxntests->raw_results(), true ); //phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
 					} else {
-						echo '<h3>' . esc_html__( 'There seems to be a problem with your site’s ability to communicate with Jetpack!', 'jetpack' ) . '</h3>';
-						echo $html;
+						$failures = $cxntests->list_fails();
+						foreach ( $failures as $fail ) {
+							echo '<div class="jetpack-test-error">';
+							echo '<p><a class="jetpack-test-heading" href="#">' . esc_html( $fail['message'] );
+							echo '<span class="noticon noticon-collapse"></span></a></p>';
+							echo '<p class="jetpack-test-details">' . esc_html( $fail['resolution'] ) . '</p>';
+							echo '</div>';
+
+							$debug_info .= "FAILED TESTS!\r\n";
+							$debug_info .= $fail['name'] . ': ' . $fail['message'] . "\r\n";
+							$debug_info .= print_r( $cxntests->raw_results(), true ); //phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
+						}
 					}
-					$debug_info .= "\r\n\r\nRAW TEST RESULTS:" . $debug_raw_info . "\r\n";
 					?>
 				</div>
-
 			<div class="entry-content">
 				<h3><?php esc_html_e( 'Trouble with Jetpack?', 'jetpack' ); ?></h3>
 				<h4><?php esc_html_e( 'It may be caused by one of these issues, which you can diagnose yourself:', 'jetpack' ); ?></h4>
 				<ol>
-					<?php /* translators: URLs to Jetpack support pages. */ ?>
-					<li><b><em><?php esc_html_e( 'A known issue.', 'jetpack' ); ?></em></b>  <?php echo sprintf( __( 'Some themes and plugins have <a href="%1$s" target="_blank">known conflicts</a> with Jetpack – check the <a href="%2$s" target="_blank">list</a>. (You can also browse the <a href="%3$s" target="_blank">Jetpack support pages</a> or <a href="%4$s" target="_blank">Jetpack support forum</a> to see if others have experienced and solved the problem.)', 'jetpack' ), 'http://jetpack.com/support/getting-started-with-jetpack/known-issues/', 'http://jetpack.com/support/getting-started-with-jetpack/known-issues/', 'http://jetpack.com/support/', 'https://wordpress.org/support/plugin/jetpack' ); ?></li>
+					<li><b><em>
+					<?php
+					/* translators: URLs to Jetpack support pages. */
```

Comment:
Not a blocker, but it seems that this is a left-over that should not be there.

Reply:
Resolved via 49dbf1704